src/text/stylized.rs

```diff
@@ -0,0 +1,224 @@
+            }
+        }
+    };
+    ($method_name_fg:ident, $method_name_bg:ident Color::$color:ident) => {
+        calculated_docs! {
+            #[doc = concat!(
+                "Sets the foreground color to [`",
+                stringify!($color),
+                "`](Color::",
+                stringify!($color),
+                ")."
+            )]
+            fn $method_name_fg(self) -> T {
+                self.color(Color::$color)
+            }
+
+            #[doc = concat!(
+                "Sets the background color to [`",
+                stringify!($color),
+                "`](Color::",
+                stringify!($color),
+                ")."
+            )]
+            fn $method_name_bg(self) -> T {
+                self.on_color(Color::$color)
+            }
+        }
+    };
+}
+
+/// The trait that enables something to be have a style.
+/// # Examples
+/// ```
+/// use ratatui::{
+///     style::{Color, Modifier, Style},
+///     text::{Stylize, Span},
+/// };
+///
+/// assert_eq!(
+///    "hello".red().on_blue().bold(),
+///     Span::styled("hello", Style::default().fg(Color::Red).bg(Color::Blue).add_modifier(Modifier::BOLD))
+/// )
+pub trait Stylize<'a, T>: Sized {
+    // Colors
+    fn color<S: Into<Color>>(self, color: S) -> T;
+    fn on_color(self, color: Color) -> T;
+
+    stylize_method!(black, on_black Color::Black);
+    stylize_method!(red, on_red Color::Red);
+    stylize_method!(green, on_green Color::Green);
+    stylize_method!(yellow, on_yellow Color::Yellow);
+    stylize_method!(blue, on_blue Color::Blue);
+    stylize_method!(magenta, on_magenta Color::Magenta);
+    stylize_method!(cyan, on_cyan Color::Cyan);
+    stylize_method!(gray, on_gray Color::Gray);
+    stylize_method!(dark_gray, on_dark_gray Color::DarkGray);
+    stylize_method!(light_red, on_light_red Color::LightRed);
+    stylize_method!(light_green, on_light_green Color::LightGreen);
+    stylize_method!(light_yellow, on_light_yellow Color::LightYellow);
+    stylize_method!(light_blue, on_light_blue Color::LightBlue);
+    stylize_method!(light_magenta, on_light_magenta Color::LightMagenta);
+    stylize_method!(light_cyan, on_light_cyan Color::LightCyan);
+    stylize_method!(white, on_white Color::White);
+
+    // Styles
+    fn reset(self) -> T;
+
+    // Modifiers
+    fn modifier(self, modifier: Modifier) -> T;
+
+    stylize_method!(bold Modifier::BOLD);
+    stylize_method!(dimmed Modifier::DIM);
+    stylize_method!(italic Modifier::ITALIC);
+    stylize_method!(underline Modifier::UNDERLINED);
+    stylize_method!(slow_blink Modifier::SLOW_BLINK);
+    stylize_method!(rapid_blink Modifier::RAPID_BLINK);
+    stylize_method!(reversed Modifier::REVERSED);
+    stylize_method!(hidden Modifier::HIDDEN);
+    stylize_method!(crossed_out Modifier::CROSSED_OUT);
+}
+
+impl<'a, T, U> Stylize<'a, T> for U
+where
+    U: Styled<Item = T>,
+{
+    fn color<S: Into<Color>>(self, color: S) -> T {
```

Comment:
I think this probably still makes sense to be named `fg`, as users will see method this in the type's autocomplete when they import Stylized. Perhaps add some docs on this?